### DISCUSSION TITLE
AddFeature - How to add features/genes to seurat object

### DISCUSSION THREAD

**Discussion by neilrobertson** [satijalab/seurat#5103] [Q&A]
Hi there,

Looking for bit of help… does anyone know how to add a new set of genes/features to a Seurat object?  I’m looking to just add in some LINE1 expression levels to an existing analysis.  I can rbind these new transcripts to the current counts matrix, but it doesn’t seem to update the object (rownames etc.) and they aren’t visible to other functions (like VlnPlot) downstream of this.  

There should be some kind of method to add genes, like: AddFeatures(seurat_object, data.frame).  Although I can understand why implementing such a feature would do more harm than good.

The alternative here is to append the LINE1 transcript counts to the main counts matrix at the very beginning, then re-run the whole analysis, but we've spent a lot of time working on this and wouldn’t want to necessarily re-do.  Any help to hack this would be much appreciated!

Thanks again,

Neil

**Comment by saketkc** [comment]
For adding new features, you need to create a new object, starting from the counts.

**Reply by timoast** [reply]
Just want to add also that if _all_ you want to do is visualize an extra gene in your dataset that for whatever reason wasn't in the original matrix, you can add it to the object metadata and then visualize its expression using `FeaturePlot`, etc. 

If you want to use the gene for other things like DE or dimension reduction it needs to be in the original matrix used to create the object

**Reply by sdambrosio** [reply]
I am interested in doing this -- how exactly would I add the extra gene to my object metadata? Thank you!

**Reply by timoast** [reply]
You can use the AddMetaData function: https://mojaveazure.github.io/seurat-object/reference/AddMetaData.html
